packages/core-data/src/actions.js

```diff
@@ -139,6 +139,70 @@ export function receiveEmbedPreview( url, preview ) {
 	};
 }
 
+/**
+ * Action triggered to delete an entity record.
+ *
+ * @param {string} kind      Kind of the deleted entity.
+ * @param {string} name      Name of the deleted entity.
+ * @param {Object} recordId  Record to be deleted.
+ */
+export function* deleteEntityRecord( kind, name, recordId ) {
+	const entities = yield getKindEntities( kind );
+	const entity = find( entities, { kind, name } );
+	if ( ! entity ) {
+		return;
+	}
+
+	yield {
+		type: 'DELETE_ENTITY_RECORD_START',
+		kind,
+		name,
+		recordId,
+	};
+
+	yield removeItems( kind, name, Number( recordId ) );
```

Comment:
Why is the ID being coerced to a number?

Reply:
As far as I can remember it is a string here but in the state it is a number (the state is keyed by this id as int) and in the reducer removing it didn't work.

Reply:
🤦 non-number IDs!